firmware/examples/PWM/source/main.cpp

```diff
@@ -0,0 +1,37 @@
+#include <cstdarg>
+#include <cstdio>
+#include <cstdint>
+#include "L0_LowLevel/delay.hpp"
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/pwm.hpp"
+#include "L2_Utilities/debug_print.hpp"
+
+int main(void)
+{
+    DEBUG_PRINT("Pwm Application Starting...\n");
+
+    // Standard gpio powered LED
+    DEBUG_PRINT("Creating Gpio powered LED...\n");
+    Gpio p4_28(4,28);
+    p4_28.SetAsInput();
+    p4_28.SetPinFunction(PinConfigureInterface::PinMode::kPullUp);
+
+    // Pwm powered LED with duty cycle sweep
+    DEBUG_PRINT("Creating Pwm powered LED...\n");
+    Pwm p2_0 = Pwm::CreatePwm<1>();
+    // Initialize Pwm at 1,000 hz
+    p2_0.Initialize(1'000);
+    float duty = 0;
+
+    while(1)
+    {
+        for(duty = 0; duty < 1; duty += 0.01)
+        {
+            p2_0.SetDutyCycle(duty);
+            Delay(100);
+        }
+    }
+
+    DEBUG_PRINT("Halting any action.\n");
```

Comment:
This line of code is unreachable, thus it shouldn't be here. Since you aren't halting action, you are looping above, you should state, above the while(1) loop that you are sweeping the duty cycle from 0 to 100%.